src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -2,520 +2,265 @@
-                // Type HMACRIPEMD160Type = typeof(System.Security.Cryptography.HMACRIPEMD160);
-                // Type MAC3DESType = typeof(System.Security.Cryptography.MACTripleDES);
-                // Type DSASignatureDescriptionType = typeof(System.Security.Cryptography.DSASignatureDescription);
-                // Type RSAPKCS1SHA1SignatureDescriptionType = typeof(System.Security.Cryptography.RSAPKCS1SHA1SignatureDescription);
-                // Type RSAPKCS1SHA256SignatureDescriptionType = typeof(System.Security.Cryptography.RSAPKCS1SHA256SignatureDescription);
-                // Type RSAPKCS1SHA384SignatureDescriptionType = typeof(System.Security.Cryptography.RSAPKCS1SHA384SignatureDescription);
-                // Type RSAPKCS1SHA512SignatureDescriptionType = typeof(System.Security.Cryptography.RSAPKCS1SHA512SignatureDescription);
-                // string RIPEMD160ManagedType = "System.Security.Cryptography.RIPEMD160Managed" + AssemblyName_Encoding;
-                // string ECDiffieHellmanCngType = "System.Security.Cryptography.ECDiffieHellmanCng, " + AssemblyName_Cng;
-                // string MD5CngType = "System.Security.Cryptography.MD5Cng, " + AssemblyName_Cng;
-                // string SHA1CngType = "System.Security.Cryptography.SHA1Cng, " + AssemblyName_Cng;
-                // string SHA256CngType = "System.Security.Cryptography.SHA256Cng, " + AssemblyName_Cng;
-                // string SHA384CngType = "System.Security.Cryptography.SHA384Cng, " + AssemblyName_Cng;
-                // string SHA512CngType = "System.Security.Cryptography.SHA512Cng, " + AssemblyName_Cng;
-                // string SHA256CryptoServiceProviderType = "System.Security.Cryptography.SHA256CryptoServiceProvider, " + AssemblyName_Csp;
-                // string SHA384CryptoSerivceProviderType = "System.Security.Cryptography.SHA384CryptoServiceProvider, " + AssemblyName_Csp;
-                // string SHA512CryptoServiceProviderType = "System.Security.Cryptography.SHA512CryptoServiceProvider, " + AssemblyName_Csp;
-                // string DpapiDataProtectorType = "System.Security.Cryptography.DpapiDataProtector, " + AssemblyRef.SystemSecurity;
-            }
-        }
-
         public static void AddAlgorithm(Type algorithm, params string[] names)
         {
             if (algorithm == null)
+            {
                 throw new ArgumentNullException(nameof(algorithm));
+            }
             if (!algorithm.IsVisible)
+            {
                 throw new ArgumentException(SR.Cryptography_AlgorithmTypesMustBeVisible, nameof(algorithm));
+            }
             if (names == null)
+            {
                 throw new ArgumentNullException(nameof(names));
+            }
 
-            string[] algorithmNames = new string[names.Length];
-            Array.Copy(names, 0, algorithmNames, 0, algorithmNames.Length);
+            string[] safeCopy = names.Length == 1 ? new string[1] { names[0] } : names.ToArray();
```

Comment:
Does `new string[1] { names[0] }` actually make that difference? If so, could we perhaps improve `ToArray` instead?

Reply:
I suspect CryptoConfig.AddAlgorithm most often called with 1 argument. We can either special case this - not create safeCopy if names.Length == 1 or even create an overload AddAlgorithm, which would take string name instead string names[].

Reply:
This is definitely not worth extra code/complexity.  Just do:
```C#
string[] algorithmNAmes = names.AsSpan().ToArray();
```
It's simpler and its perf will be just fine; I cannot believe any perf difference will be measurable in real-world usage, especially considered we're about to do cryptography!

(If you really believed Length == 1 was special and warranted optimization, you'd want to special case everything for the rest of the method, too; there's no point allocating another array, looping with foreach twice, etc., when you can just operate on the sole value.)